unity-renderer/Assets/DCLPlugins/ECS7/ECSComponents/VideoPlayer/VideoPlayerHandler.cs

```diff
@@ -50,6 +59,8 @@ public void OnComponentRemoved(IParcelScene scene, IDCLEntity entity)
             Helpers.Utils.OnCursorLockChanged -= OnCursorLockChanged;
             loadingScreen.visible.OnChange -= OnLoadingScreenStateChanged;
 
+            componentWriter.RemoveComponent(scene, entity, ComponentID.VIDEO_EVENT);
```

Comment:
This might lead to a wrong state, since component is also written when `ECSVideoPlayerSystem` is running, we should probably write the removal of the component on that system, you could add a `removed` property on the `videoEventInternalComponent` or create a "tag" component to handle that

Reply:
To reach a state of "adding and removing the video event component only in the video player system" I'm still working on actually implement the `removed` property in the `InternalVideoPlayer` component and avoid removing it in the `VideoPlayerHander.OnComponentRemoved()`, because otherwise, in the `Update()` of the `ECSVideoPlayerSystem` the entities with their `InternalVideoPlayer` component removed, wouldn't get to be checked and wouldn't get to clean their video event inside the system... currently I have that local implementation without errors but the video events stop working on that approach, I'm still debugging.

Reply:
Finally got it working, check this commit: https://github.com/decentraland/unity-renderer/pull/5042/commits/136dea177ffc607689a8d06060c8bc864e18e981